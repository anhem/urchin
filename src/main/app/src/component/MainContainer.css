#main-container {
    display: table-row;
    height: 100vh;
}

#main-container__main-menu {
    width: 10em;
    background-color: #4b4b4b;
    display: table-cell;
    text-align: center;
    vertical-align: top;
    margin: 0;
    padding-top: 1em;
}

#main-container__content {
    display: table-cell;
    padding-left: 1em;
    padding-top: 1em;
    vertical-align: top;
}

#main-container__content a {
    text-decoration: none;
    color: blue;
}