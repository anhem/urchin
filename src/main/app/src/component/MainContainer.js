import React, {Component} from 'react'
import {Redirect, Route, Switch} from 'react-router-dom'
import Users from "./users/Users";
import Groups from "./groups/Groups";
import MainMenu from "./MainMenu";
import Home from "./Home";
import Folders from "./folders/Folders";

class MainContainer extends Component {
    render() {
        return (
            <div className="main-container">
                <MainMenu/>
                <Switch>
                    <Route exact path="/" component={Home}/>
                    <Route path="/users" component={Users}/>
                    <Route path="/groups" component={Groups}/>
                    <Route path="/folders" component={Folders}/>
                    <Redirect to="/"/>
                </Switch>
            </div>
        );
    }
}

export default MainContainer;