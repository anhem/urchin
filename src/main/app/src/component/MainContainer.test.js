import React from 'react';
import {Router} from 'react-router-dom'
import {mount} from 'enzyme'
import MainContainer from "./MainContainer";
import history from '../history'

describe('MainContainer', () => {

    it('renders without crashing', () => {
        let component = mount(
            <Router history={history}>
                <MainContainer/>
            </Router>
        );

        expect(component.length).toEqual(1);
    });
});