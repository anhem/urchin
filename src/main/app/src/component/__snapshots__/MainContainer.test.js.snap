// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer match snapshot 1`] = `
<div
  className="main-container"
>
  <MainMenu />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/users"
    />
    <Route
      component={[Function]}
      path="/groups"
    />
    <Route
      component={[Function]}
      path="/folders"
    />
    <Redirect
      push={false}
      to="/"
    />
  </Switch>
</div>
`;
